host/volume/api/http.go

```diff
@@ -0,0 +1,114 @@
+package volumeapi
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/julienschmidt/httprouter"
+	"github.com/flynn/flynn/host/volume"
+	"github.com/flynn/flynn/host/volume/zfs"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/random"
+)
+
+type HttpAPI struct {
+	vman *volume.Manager
+}
+
+func NewHttpAPI(vman *volume.Manager) *HttpAPI {
+	return &HttpAPI{vman: vman}
+}
+
+func RegisterRoutes(api *HttpAPI, r *httprouter.Router) {
+	r.POST("/volume/provider", api.CreateProvider)
```

Comment:
Unsure.  It's fairly inexpensive in code complexity to provide it since it's effectively the same code as needed in host bootstrap, so I tilted towards inclusion.